Zend/zend_compile.c

```diff
@@ -10604,6 +10604,14 @@ static void zend_eval_const_expr(zend_ast **ast_ptr) /* {{{ */
 		return;
 	}
 
+#ifdef ZEND_CHECK_STACK_LIMIT
+	if (UNEXPECTED(zend_call_stack_overflowed(EG(stack_limit)))) {
+		/* Abort evaluation due to stack usage. Note that compiling this
+		 * expression is likely to also fail. */
```

Comment:
Not sure if this is a real problem but `zend_eval_const_expr` evaluates some AST nodes that must not appear again in constant expressions, e.g. magic constants. It might be better to fatal error right here too, in case the expression evaluation uses fewer stack frames than `zend_eval_const_expr`.